### PR TITLE
Add StreamOffset and StreamOffsets models for Stream Api

### PR DESCRIPTION
**Pull request by mykolamalik**
Models were generated using Swagger from OLP Stream API v2 and adapted
to be consistent with existing models for other layers and with Stream
Layer Client Detailed Design proposal.

Relates-To: OLPEDGE-1196

Signed-off-by: Mykola Malik <ext-mykola.malik@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  int64_t GetOffset() const { return offset_; };
+  void SetOffset(int64_t value) { offset_ = value; }
+
+ private:
+  /**
+   * @brief The partition of the stream layer for which this offset applies. It
+   * is not the same as [Partitions
+   * object](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/layers/partitions.html).
+   */
```
[the remark this answers is not in the document]
Comment (mykolamalik) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h]: done

**Comment by mykolamalik** [comment]
> 
> 
> Were is the parser? Don't we need a parser?

You're right, we need it. It is included in this review now.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  StreamOffset() = default;
+
+  int32_t GetPartition() const { return partition_; }
+  void SetPartition(int32_t value) { partition_ = value; }
+
+  int64_t GetOffset() const { return offset_; };
+  void SetOffset(int64_t value) { offset_ = value; }
+
+ private:
+  /// @brief The partition of the stream layer for which this offset applies. It
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h]: You don't need brief.

```suggestion
  /// The partition of the stream layer for which this offset applies. It
```

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  /// is not the same as [Partitions
+  /// object](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/layers/partitions.html).
+  int32_t partition_;
+
+  /// @brief The offset in the partition of the stream layer.
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h]:
```suggestion
  /// The offset in the partition of the stream layer.
```

Comment (mykolamalik) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h]: Done.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h (hunk 2 above)
Comment (mykolamalik) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/StreamOffsets.h]: Done.

**Pull request merged**